src/fmt/writer/termcolor/shim_impl.rs

```diff
@@ -30,14 +46,19 @@ impl BufferWriter {
     pub(in crate::fmt::writer) fn print(&self, buf: &Buffer) -> io::Result<()> {
         // This impl uses the `eprint` and `print` macros
         // instead of using the streams directly.
-        // This is so their output can be captured by `cargo test`
-        let log = String::from_utf8_lossy(&buf.0);
-
+        // This is so their output can be captured by `cargo test`.
         match self.target {
-            Target::Stderr => eprint!("{}", log),
-            Target::Stdout => print!("{}", log),
+            // Safety: If the target type is `Pipe`, `target_pipe` will always be non-empty.
+            TargetType::Pipe => self
+                .target_pipe
+                .as_ref()
+                .unwrap()
+                .lock()
+                .unwrap()
+                .write_all(&buf.0)?,
+            TargetType::Stdout => print!("{}", String::from_utf8_lossy(&buf.0)),
+            TargetType::Stderr => eprint!("{}", String::from_utf8_lossy(&buf.0)),
         }
-
```

Comment:
Maybe put back the empty line before the final expression of the function.

Reply:
That one just got lost ^^